lib/matplotlib/tests/test_lines.py

```diff
@@ -194,3 +197,24 @@ def test_nan_is_sorted():
 def test_step_markers(fig_test, fig_ref):
     fig_test.subplots().step([0, 1], "-o")
     fig_ref.subplots().plot([0, 0, 1], [0, 1, 1], "-o", markevery=[0, 2])
+
+
+def test_marker_as_markerstyle():
+    fig, ax = plt.subplots()
+    line, = ax.plot([2, 4, 3], marker=MarkerStyle("D"))
+    fig.canvas.draw()
+    assert line.get_marker() == "D"
+
+    #continue with smoke tests:
+    line.set_marker("s")
+    fig.canvas.draw()
+    line.set_marker(MarkerStyle("o"))
+    fig.canvas.draw()
+    # test Path roundtrip
+    triangle1 = Path([[-1., -1.], [1., -1.], [0., 2.], [0., 0.]],
+                     [Path.MOVETO, Path.LINETO, Path.LINETO, Path.CLOSEPOLY])
```

Comment:
In line with #16617:
```suggestion
    triangle1 = Path([[-1, -1], [1, -1], [0, 2], [0, 0]], closed=True)
```